### PR TITLE
ceph: remove mimic support, default to nautilus

### PR DESCRIPTION
**Pull request by leseb**
**Description of your changes:**

As of 1.3, Rook will only support Ceph Nautilus 14.2.5.

Signed-off-by: Sébastien Han <seb@redhat.com>

**Checklist:**

- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/master/development-flow.html#submitting-a-pull-request)
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.
- [ ] Pending release notes updated with breaking and/or notable changes, if necessary.
- [ ] Upgrade from previous release is tested and upgrade user guide is updated, if necessary.
- [ ] Code generation (`make codegen`) has been run to update object specifications, if necessary.
- [ ] Comments have been added or updated based on the standards set in [CONTRIBUTING.md](https://github.com/rook/rook/blob/master/CONTRIBUTING.md#comments)
- [ ] Add the flag for skipping the CI if this PR does not require a build. See [here](https://github.com/rook/rook/blob/master/INSTALL.md#skip-ci) for more details.

[test ceph]
[test full]

### CODE REVIEW

File: pkg/daemon/ceph/config/config.go [hunk 1]
```diff
@@ -335,3 +290,37 @@ func msgrPrefix(currentMonPort int32) string {
...
+		// This basically handles the transition between monitors running on 6790 to msgr2
+		// So whatever the previous monitor port was we keep it
+		currentMonPort := cephutil.GetPortFromEndpoint(monitor.Endpoint)
+
+		monPorts := [2]string{strconv.Itoa(int(Msgr2port)), strconv.Itoa(int(currentMonPort))}
+		msgr2Endpoint := net.JoinHostPort(monIP, monPorts[0])
+		msgr1Endpoint := net.JoinHostPort(monIP, monPorts[1])
+
+		monHosts[i] = "[v2:" + msgr2Endpoint + ",v1:" + msgr1Endpoint + "]"
+		// podName := os.Getenv("POD_NAME")
```
Comment (travisn) [inline pkg/daemon/ceph/config/config.go]: this commented section can be removed, right?

File: pkg/operator/ceph/cluster/osd/spec.go [hunk 2]
```diff
@@ -249,16 +247,13 @@ func (c *Cluster) makeDeployment(osdProps osdProperties, osd OSDInfo, provisionC
 	}
 
 	// As of Nautilus Ceph auto-tunes its osd_memory_target on the fly so we don't need to force it
-	if !c.clusterInfo.CephVersion.IsAtLeastNautilus() && !c.resources.Limits.Memory().IsZero() {
+	if !c.resources.Limits.Memory().IsZero() {
```
Comment (SerialVelocity) [inline pkg/operator/ceph/cluster/osd/spec.go:250]: @leseb This block should've been removed, right? The old code was only applying it if the version was below Nautilus.

Comment (leseb) [reply pkg/operator/ceph/cluster/osd/spec.go:250]: Good catch, let me fix this.

Comment (leseb) [reply pkg/operator/ceph/cluster/osd/spec.go:250]: See: https://github.com/rook/rook/pull/5529

Comment (leseb) [reply pkg/operator/ceph/cluster/osd/spec.go:250]: Expect a backport soon

**Pull request merged**